File: naives.go
package main

import (
	"fmt"
	"strings"
)

var reference = "AGCATCGATCGATCGATCGATCGATTGTCGATCGATCGATGT"
var pattern = "AGT"

func naive(ref string, patt string, thresh int) {
	occ := []int{}
	for i := 0; i < (len(ref) - len(patt) + 1); i++ {
		var mismatch = 0
		var match = true
		for j := 0; j < len(patt); j++ {
			if ref[i+j] != patt[j] {
				mismatch = mismatch + 1
				if mismatch > thresh {
					match = false
					break
				}
			}
		}
		if match {
			occ = append(occ,i)
		}
	}
	fmt.Println(strings.Trim(strings.Replace(fmt.Sprint(occ), " ", ",", -1), "[]"))
}

func main() {
	naive(reference,pattern,1)
}

File: naives.ts
var reference:string = "AGCATCGATCGATCGATCGATCGATTGTCGATCGATCGATGT"
var pattern:string  = "AGT"

function naive(ref, patt, thresh) {
	var occ:number[] = []
	for (var i:number = 0; i < (ref.length-patt.length+1); i++) {
		var mismatch:number = 0
		var match:boolean = true
		for (var j:number = 0; j < patt.length; j++) {
			if (ref[i+j] != patt[j]) {
				mismatch++
				if (mismatch > thresh) {
					match = false
					break
				}
			}
		}
		if (match) {
			occ.push(i)
		}
	}
	return occ
}

console.log(naive(reference, pattern, 1).join(","))
